File: src/WWS/User/Web/IntegrationEvents/CategoryPromotionCreatedEvent.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Web.IntegrationEvents
{
    public class CategoryPromotionCreatedEvent : ICategoryPromotionCreatedEvent
    {
        public string CategoryName { get; set; }
        public int Discount { get; set; }
        public string StartDate { get; set; }
        public string EndDate { get; set; }
        public string email { get; set; }
    }

    public interface ICategoryPromotionCreatedEvent
    {
        public string CategoryName { get; set; }
        public int Discount { get; set; }
        public string StartDate { get; set; }
        public string EndDate { get; set; }
        public string email { get; set; }
    }
}

File: src/WWS/Mail/mail/src/main/kotlin/hu/bme/aut/mail/integrationEvent/event/CategoryPromotionCreatedEvent.kt
package hu.bme.aut.mail.integrationEvent.event

import com.fasterxml.jackson.annotation.JsonProperty
import java.io.Serializable

data class CategoryPromotionCreatedEvent(
    @JsonProperty("categoryName") val categoryName: String,
    @JsonProperty("discount") val discount: Int,
    @JsonProperty("startDate") val startDate: String,
    @JsonProperty("endDate") val endDate: String,
    @JsonProperty("email") val email: String
) : Serializable